cocos/game/game.ts

```diff
@@ -518,6 +519,27 @@ export class Game extends EventTarget {
         director.tick(this.frameTime / 1000);
     }
 
+    /**
+     * @en Called by the engine to pause the game.
+     * @zh 提供给引擎调用暂停游戏接口。
+     */
+    private pauseByEngine () {
+        if (this._paused) { return; }
```

Comment:
Why not check this._pausedByEngine instead?

Reply:
This function is called by the engine, and the engine calls it every time it switches the background. The _paused check is to avoid restoring the 'pause' called by the user.

Reply:
> The _paused check is to avoid restoring the 'pause' called by the user.

Could you please describe it in detail? What's error if not check it?

Reply:
It is not necessary to check _pausedByEngine, because it is called and restored every time the background is switched, and checking _paused is necessary to distinguish it from the pause called by the user

Reply:
Seems reasonable.